src/symbolize/mod.rs

```diff
@@ -78,7 +79,7 @@ impl Symbol {
     /// unix platforms other than OSX) and when a binary is compiled with
     /// debuginfo. If neither of these conditions is met then this will likely
     /// return `None`.
-    pub fn filename(&self) -> Option<&Path> {
+    pub fn filename(&self) -> Option<Cow<Path>> {
```

Comment:
This is unfortunately a breaking change for this crate, but I would it be possible to avoid this perhaps? Could some unsafe code be used to store the owned version of a file, if necessary, on the windows implementation?

Reply:
Oops, i didn't realize this is for user...  to keep the original signature, i have store the OsString/PathBuf somewhere, do you think i can add an extra field in the Symbol struct for this case? Or should i collect them together as something like a global static variable?

Reply:
Yeah adding a field to the `Symbol` struct should be fine as that's all private implementation details anyway